fasta2a/fasta2a/worker.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import annotations as _annotations
+
+from abc import ABC, abstractmethod
+from contextlib import AsyncExitStack
+from dataclasses import dataclass
+from typing import Annotated, Any, Generic, Literal, TypeVar
+
+import anyio
+from opentelemetry.trace import Span, get_current_span, get_tracer, use_span
+from pydantic import Discriminator
+from typing_extensions import Self, TypedDict
+
+from .runner import Runner
+from .schema import TaskIdParams, TaskSendParams
+from .storage import Storage
+
+tracer = get_tracer(__name__)
+
+
+class WorkerClient(ABC):
+    """A client for the worker."""
+
+    @abstractmethod
+    async def send_run_task(self, params: TaskSendParams) -> None:
+        """Send a task to be executed by the worker."""
+        raise NotImplementedError('send_run_task is not implemented yet.')
+
+    @abstractmethod
+    async def send_cancel_task(self, params: TaskIdParams) -> None:
+        """Cancel a task."""
+        raise NotImplementedError('send_cancel_task is not implemented yet.')
+
+
+class WorkerServer(ABC):
+    """A server for the worker."""
+
+    @abstractmethod
+    async def run_task(self, params: TaskSendParams) -> None:
+        """Run a task."""
+        raise NotImplementedError('run_task is not implemented yet.')
+
+    @abstractmethod
+    async def cancel_task(self, params: TaskIdParams) -> None:
+        """Cancel a task."""
+        raise NotImplementedError('cancel_task is not implemented yet.')
+
+
+@dataclass
+class Worker(WorkerClient, WorkerServer, ABC):
+    """The worker class is in charge of scheduling and running tasks.
```

Comment:
>     class Runner(ABC):
>         """A runner is responsible for executing tasks."""

So the `Worker` _runs_ tasks, and the `Runner` _executes_ tasks? Something seems wrong here, or at best confusing.

Reply:
It seems to me the Worker doesn't _run_ the tasks, it just hands them off to the runner to be run. I think we should rework the naming a bit..

Reply:
Can we rename `Worker` to `Scheduler`? That would address the issue for me.